File: compressao-de-cados/Python/matriz/dado.py
class Dado:

    def __init__(self, valor: int, lin: int, col: int):
        """_summary_: construtor da classe Dado

        Args:
            valor (int): _description_: valor inteiro armazenado na célula da matriz
            lin (int): _description_: índice da linha onde o valor se encontra na matriz
            col (int): _description_: índice da coluna onde o valor se encontra na matriz
        """
        self.valor = valor
        self.lin = lin
        self.col = col

    def __eq__(self, other):
        """_summary_: compara dois objetos Dado pela linha e coluna, ignorando o valor

        Args:
            other (_type_): _description_: objeto a ser comparado

        Returns:
            _type_: _description_: True se lin e col forem iguais, False caso contrário
        """
        if self is other:
            return True
        if other is None or not isinstance(other, Dado):
            return False
        return self.lin == other.lin and self.col == other.col

    def __repr__(self):
        """_summary_: retorna uma representação textual do objeto Dado

        Returns:
            _type_: _description_: string no formato Dado{valor=X, lin=Y, col=Z}
        """
        return f"Dado{{valor={self.valor}, lin={self.lin}, col={self.col}}}"

File: compressao-de-dados/Java/matriz/Dado.java
public class Dado {
  public int valor;
  public int lin;
  public int col;

  public Dado(int valor, int lin, int col) {
    this.valor = valor;
    this.lin = lin;
    this.col = col;
  }

    @Override 
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Dado other = (Dado) obj;
        if (this.lin != other.lin) {
            return false;
        }
        return this.col == other.col;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Dado{");
        sb.append("valor=").append(valor);
        sb.append(", lin=").append(lin);
        sb.append(", col=").append(col);
        sb.append('}');
        return sb.toString();
    }
}
